rock_paper_scissors/lib/rock_paper_scissors.ex

```diff
@@ -18,6 +18,18 @@ defmodule RockPaperScissors do
       iex> RockPaperScissors.score({:p1, :scissors}, {:p2, :rock})
       :p2
   """
-  def score(t1, t2) do
+
+  permutations = [{:rock, :scissors}, {:scissors, :paper}, {:paper, :rock}]
+  for {winner, loser} <- permutations do
+    defp get_winner(unquote(winner), unquote(loser)), do: :p1
+    defp get_winner(unquote(loser), unquote(winner)), do: :p2
```

Comment:
Have you considered defining only the winning cases and then having a catch-all function with a signature like `get_winner(_, _)`. That's what I did, since the other function clauses already took care of draw and winning.